app/src/main/java/org/mozilla/fenix/settings/logins/controller/SavedLoginsStorageController.kt

```diff
@@ -49,7 +49,7 @@ open class SavedLoginsStorageController(
             }
             deleteLoginJob?.await()
             withContext(Dispatchers.Main) {
-                navController.popBackStack(R.id.savedLoginsFragment, false)
+                navController.navigate(R.id.action_loginDetailFragment_to_savedLoginsFragment)
```

Comment:
This will recreate the fragment which will recreate the `savedLoginsStore` and as such show the up-to-date list of logins.
It is a solution for the problem at hand but seems a bit excessive and goes around our MVI.

It seems like the issue is happening because [we are using `StoreProvider` to cache our Store in between accessing the `SavedLoginsFragment`](https://github.com/mozilla-mobile/fenix/blob/02d4d59358117a4cb21ce661ae113107af98375f/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/SavedLoginsFragment.kt#L69-L73) and the [store does not currently support a delete action](https://github.com/mozilla-mobile/fenix/blob/02d4d59358117a4cb21ce661ae113107af98375f/app/src/main/java/org/mozilla/fenix/settings/logins/LoginsFragmentStore.kt#L53-L61) needed to actually update the state to contain one less login after deletion.

So we need to add support for this new action and call it in the controller before going back to the list of saved logins.
This action would also be useful if later we decide to show a menu / button to support deletion directly in `SavedLoginsFragment` like we do for other items shown as lists like bookmarks, history.
cc @topotropic about this suggestion.